syntax = "proto3";

option java_multiple_files = true;
option java_package = "com.kvstore.grpc";
option java_outer_classname = "KVStoreProto";

package kvstore;

service KVStoreService {
  rpc Get(GetRequest) returns (GetResponse) {}
  rpc Put(PutRequest) returns (PutResponse) {}
}

message GetRequest {
  string key = 1;
}

message GetResponse {
  string value = 1;
  bool found = 2;
}

message PutRequest {
  string key = 1;
  string value = 2;
}

message PutResponse {
  bool success = 1;
}